exporter/elasticsearchexporter/exporter_test.go

```diff
@@ -349,14 +349,10 @@ func TestExporterLogs(t *testing.T) {
 
 	t.Run("no retry", func(t *testing.T) {
 		configurations := map[string]func(*Config){
-			"max_requests limited": func(cfg *Config) {
```

Comment:
[to reviewer] surprisingly, setting retry::enabled to true and retry::max_requests to 1 in the past will cause 0 to be passed to es client and result in a default of 3 retries. This can be seen as a bug but will be out of scope of this PR. Therefore, I've clarified in the README that user should set retry::enabled to false instead of retry::max_retries to 0

Reply:
Oooh thanks for the clarification :+1:

Reply:
In this case, shouldn't we treat `retry::max_retries: 0` as invalid when `retry::enabled` is `true`?

Reply:
I don't think it will ever be 0, because if it is unspecified or 0, we will make it the default = 2. So I wonder how we would ever enforce this

Reply:
Oh, right, the `retry::max_retries` is defined as an `int` and not `*int`, so we won't be able to tell in the `Validate()` method whether the user explicitly set it to `0` or they haven't set it at all. I guess we could either change the type of `retry::max_retries` to `*int`, or just leave it as is and not validate this. I'd rather go with the first option, but the second one is acceptable too, as Curtis [commented](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35571/files#r1786639764).